config/notify.yml

```diff
@@ -1,14 +1,18 @@
+<% fake_api_key = "fake_api_key-00000000-0000-0000-0000-000000000000-00000000-0000-0000-0000-000000000000" %>
+<% fake_base_url = "http://fake-notify.com" %>
+<% template_id = "cb633abc-6ae6-4843-ae6f-82ca500b6de2" %>
+
 development:
-  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] %>
-  template_id: <%= ENV['GOVUK_NOTIFY_TEMPLATE_ID'] || "cb633abc-6ae6-4843-ae6f-82ca500b6de2" %>
-  base_url: <%= ENV['GOVUK_NOTIFY_BASE_URL'] %>
+  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] || fake_api_key %>
+  base_url: <%= ENV['GOVUK_NOTIFY_BASE_URL'] || fake_base_url %>
+  template_id: <%= ENV['GOVUK_NOTIFY_TEMPLATE_ID'] || template_id %>
 
 test:
-  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] %>
-  template_id: <%= ENV['GOVUK_NOTIFY_TEMPLATE_ID'] || "cb633abc-6ae6-4843-ae6f-82ca500b6de2" %>
-  base_url: <%= ENV['GOVUK_NOTIFY_BASE_URL'] %>
+  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] || fake_api_key %>
+  base_url: <%= ENV['GOVUK_NOTIFY_BASE_URL'] || fake_base_url %>
+  template_id: <%= ENV['GOVUK_NOTIFY_TEMPLATE_ID'] || template_id %>
 
 production:
-  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] %>
-  template_id: <%= ENV['GOVUK_NOTIFY_TEMPLATE_ID'] || "cb633abc-6ae6-4843-ae6f-82ca500b6de2" %>
-  base_url: <%= ENV['GOVUK_NOTIFY_BASE_URL'] %>
+  api_key: <%= ENV['GOVUK_NOTIFY_API_KEY'] || fake_api_key %>
```

Comment:
I think in production it would be better that we can't instantiate the Notify client if we're missing credentials. In fact, I would be tempted to use `ENV.fetch` for all of these, what do you think?

Reply:
@thomasleese updated